### PR TITLE
reVUE backend integration

### PR DESCRIPTION
**Pull request by aditigopalan**
Integrating list of VUEs from www.cancerrevue.org into the backend server of Genome Nexus. 

It updates and fixes the revised protein effect for all variants that are VUEs. However, some VUE information like protein changes will need to be updated in the future.

### CODE REVIEW

File: data/curiousCases/VUEs.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            "revisedProteinEffect": "p.549_557del", 
+            "variantClassification":"In_Frame_Del"
+        },
+        {
+            "variant":"4:g.55593576_55593608del",
+            "genomicLocation": "4,5,55593576,55593608,ATATAATATATATATAT,-", 
+            "transcriptId": "ENST00000288137", 
+            "revisedProteinEffect": "p.549_557del", 
+            "variantClassification":"In_Frame_Del"
+        }
```
Comment (leexgh) [inline data/curiousCases/VUEs.json:32]: Good to remove these vues that we made up randomly 😅

File: data/curiousCases/VUEs.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        "genomicLocation": "Deletions flanking intron 10 - exon 11 boundary (4:55593576_55593606del)",
+        "defaultEffect": "splice",
+        "comment": "Changes splice donor site, cause inframe deletion in exon 11",
+        "context": "Actionable in GIST",
+        "pubmedIds": [15507676],
+        "referenceText": "Corless et al, 2004",
+        "revisedProteinEffects": [ 
+        {
+            "variant":"4:g.55593576_55593606del", 
+            "genomicLocation": "4,5,55593576,55593606,ATATAATATATATATAT,-",
```
Comment (leexgh) [inline data/curiousCases/VUEs.json:14]: Sorry this is just an example string I put here, the real genomic location should be: `4,5,55593576,55593606,CCACAGAAACCCATGTATGAAGTACAGTGGA,-`

File: model/src/main/java/org/cbioportal/genome_nexus/model/VUEs.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    private String transcriptId;
+    private String genomicLocation;
+    private int[] pubmedIds;     
+    private String context;
+    private String comment;
+    private String referenceText;
+    private String defaultEffect;          
+    private List<RevisedProteinEffect> revisedProteinEffects;
+
+    public String gethugoGeneSymbol() {
```
Comment (leexgh) [inline model/src/main/java/org/cbioportal/genome_nexus/model/VUEs.java:17]:
```suggestion
    public String getHugoGeneSymbol() {
```
As well as all getter and setter functions in this file. Could simply delete all and re-generate from vscode

File: service/src/main/java/org/cbioportal/genome_nexus/service/internal/VariantAnnotationSummaryServiceImpl.java [hunk 4]
```diff
@@ -211,6 +223,7 @@ private TranscriptConsequenceSummary getTranscriptSummary(VariantAnnotation anno
                                                               TranscriptConsequence transcriptConsequence)
     {
         TranscriptConsequenceSummary summary = null;
+        System.out.println(vuesList);
```
Comment (leexgh) [inline service/src/main/java/org/cbioportal/genome_nexus/service/internal/VariantAnnotationSummaryServiceImpl.java:226]: Remove print functions

File: service/src/main/java/org/cbioportal/genome_nexus/util/jsonReader.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+        Gson gson = new Gson();
+
+        try (Reader reader = new FileReader("data/curiousCases/VUEs.json")) {
+
+            // Convert JSON File to Java Object
+            VUEs [] vues = gson.fromJson(reader, VUEs[].class);
+			
+			// print vues 
+            System.out.println(vues);
```
Comment (leexgh) [inline service/src/main/java/org/cbioportal/genome_nexus/util/jsonReader.java:22]: Same here we don't need print

File: service/src/main/java/org/cbioportal/genome_nexus/service/internal/VariantAnnotationSummaryServiceImpl.java [hunk 6]
```diff
@@ -235,6 +248,25 @@ private TranscriptConsequenceSummary getTranscriptSummary(VariantAnnotation anno
             summary.setPolyphenScore(transcriptConsequence.getPolyphenScore());
             summary.setSiftPrediction(transcriptConsequence.getSiftPrediction());
             summary.setSiftScore(transcriptConsequence.getSiftScore());
+            
+            System.out.println(vuesList);
```
Comment (leexgh) [inline service/src/main/java/org/cbioportal/genome_nexus/service/internal/VariantAnnotationSummaryServiceImpl.java:252]: Same here

File: service/src/main/java/org/cbioportal/genome_nexus/service/internal/VariantAnnotationSummaryServiceImpl.java [hunk 7]
```diff
@@ -20,8 +21,17 @@
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
```
Comment (leexgh) [inline service/src/main/java/org/cbioportal/genome_nexus/service/internal/VariantAnnotationSummaryServiceImpl.java:25]: HashMap is not used, right? we can remove unused import

**Review by leexgh — CHANGES_REQUESTED** [review]

**Review by leexgh — APPROVED** [review]

File: service/src/main/java/org/cbioportal/genome_nexus/service/internal/VariantAnnotationSummaryServiceImpl.java [hunk 8]
```diff
@@ -235,6 +246,21 @@ private TranscriptConsequenceSummary getTranscriptSummary(VariantAnnotation anno
...
+
+            List<VUEs> vueArray =  Arrays.asList(vuesList);
+            Map <String, RevisedProteinEffect> vuesMap = vueArray
+            .stream()
+            .map(VUEs::getRevisedProteinEffects)
+            .filter(Objects::nonNull)
+            .flatMap(revisedProteinEffects -> revisedProteinEffects.stream())
+            .collect(Collectors.toMap(RevisedProteinEffect::getTranscriptId, Function.identity()));
+
+            if (vuesMap.get(transcriptConsequence.getTranscriptId()) != null && vuesMap.get(transcriptConsequence.getTranscriptId()).getVariant().equals(annotation.getVariant()))
```
Comment (inodb) [inline service/src/main/java/org/cbioportal/genome_nexus/service/internal/VariantAnnotationSummaryServiceImpl.java:258]: does this need a null check for when vuesMap returns nothing?

**Pull request merged**